ui-admin/src/study/StudySettings.tsx

```diff
@@ -1,39 +1,79 @@
-import React from 'react'
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
-import { faEdit } from '@fortawesome/free-solid-svg-icons'
+import React, { useState } from 'react'
 import { StudyEnvContextT } from './StudyEnvironmentRouter'
 import { LoadedPortalContextT } from '../portal/PortalProvider'
 import PortalEnvConfigView from '../portal/PortalEnvConfigView'
 import { PortalEnvironment } from '@juniper/ui-core'
+import { doApiLoad } from 'api/api-utils'
+import { Button } from 'components/forms/Button'
+import { set } from 'lodash/fp'
+import { useUser } from 'user/UserProvider'
+import Api from 'api/api'
+import { Store } from 'react-notifications-component'
+import { successNotification } from 'util/notifications'
+import LoadingSpinner from 'util/LoadingSpinner'
 
 /** shows settings for both a study and its containing portal */
 export default function StudySettings({ studyEnvContext, portalContext }:
 {studyEnvContext: StudyEnvContextT, portalContext: LoadedPortalContextT}) {
-  const envConfig = studyEnvContext.currentEnv.studyEnvironmentConfig
   const portalEnv = portalContext.portal.portalEnvironments
     .find(env =>
       env.environmentName === studyEnvContext.currentEnv.environmentName) as PortalEnvironment
-  return <div className="container bg-white">
+  return <div className="ps-4">
+    <StudyEnvConfigView studyEnvContext={studyEnvContext} portalContext={portalContext}/>
+    <PortalEnvConfigView portalEnv={portalEnv} portalContext={portalContext}/>
+  </div>
+}
+
+/** allows editing config settings for a particular study */
+export function StudyEnvConfigView({ studyEnvContext, portalContext }:
+                                       {studyEnvContext: StudyEnvContextT, portalContext: LoadedPortalContextT}) {
+  const [config, setConfig] = useState(studyEnvContext.currentEnv.studyEnvironmentConfig)
+  const { user } = useUser()
+  const [isLoading, setIsLoading] = useState(false)
+  /** update a given field in the config */
+  const updateConfig = (propName: string, value: string | boolean) => {
+    setConfig(set(propName, value))
+  }
+
+  /** saves any changes to the server */
+  const save = async (e: React.MouseEvent) => {
+    e.preventDefault()
```

Comment:
The `Button` component specifies `type="button"` which should prevent the form submit behavior without this line. (if that's what you were preventing here)